src/d3a/d3a_core/sim_results/area_statistics.py

```diff
@@ -592,3 +591,41 @@ def _convert_output_format(self, price_energy, csv_output, redis_output):
                 } for timeslot, trades in trade_rates.items()
             ]
             redis_output[node.uuid] = deepcopy(csv_output[node.name])
+
+
+class BaselinePeakEnergyStats:
+    def __init__(self):
+        self.baseline_peak_percentage_result = {}
+        self.baseline_peak_percentage_result_redis = {}
+
+    def update(self, area):
+        self.update_exported_imported_energy(area)
+        self.results_to_redis(area)
+
+    def update_exported_imported_energy(self, area):
+        if area.baseline_energy_settings is not None:
+            area.stats.aggregate_exported_imported_energy(area)
+            baseline_import = area.baseline_energy_settings["baseline_peak_energy_import_kWh"]
+            baseline_export = area.baseline_energy_settings["baseline_peak_energy_export_kWh"]
+            if baseline_import > 0:
+                create_subdict_or_update(self.baseline_peak_percentage_result, area.name,
+                                         {"import": {k: v/baseline_import
+                                          for k, v in area.stats.imported_energy.items()}})
+            if baseline_export > 0:
+                create_subdict_or_update(self.baseline_peak_percentage_result, area.name,
+                                         {"export": {k: v/baseline_export
+                                          for k, v in area.stats.exported_energy.items()}})
+        for child in area.children:
+            if child.strategy is None:
+                self.update_exported_imported_energy(child)
+
+    def results_to_redis(self, area):
+        if area.name in self.baseline_peak_percentage_result:
+            rounded_result = {}
+            for key in self.baseline_peak_percentage_result[area.name].keys():
+                rounded_result[key] = \
+                    {k: round_floats_for_ui(v)
+                     for k, v in self.baseline_peak_percentage_result[area.name][key].items()}
+            self.baseline_peak_percentage_result_redis[area.uuid] = rounded_result
+        for child in area.children:
+            self.results_to_redis(child)
```

Comment:
This additional recursion could potentially be  saved if `self.baseline_peak_percentage_result_redis` is populated right after the calculation of `self.baseline_peak_percentage_result`

Reply:
you are right! Will move this into `update_exported_imported_energy`